webapp/src/main/java/com/box/l10n/mojito/service/smartling/ThirdPartyTextUnitSearchService.java

```diff
@@ -0,0 +1,130 @@
+        this.importerCacheService = importerCacheService;
+        this.thirdPartyTextUnitRepository = thirdPartyTextUnitRepository;
+    }
+
+    String pluralSuffix = "_other";
+
+    /**
+     * logger
+     */
+    static Logger logger = LoggerFactory.getLogger(ThirdPartyTextUnitSearchService.class);
+
+    List<ThirdPartyTextUnitForBatchImport> convertDTOToBatchImport(Set<ThirdPartyTextUnitDTO> thirdPartyTextUnitDTOSet, boolean isPluralFile) {
+
+        logger.debug("Create caches to map convert to ThirdPartyTextUnitForBatchImport list");
+        LoadingCache<String, Repository> repositoriesCache = importerCacheService.createRepositoriesCache();
+        LoadingCache<Map.Entry<String, Long>, Asset> assetsCache = importerCacheService.createAssetsCache();
+        LoadingCache<Map.Entry<String, Long>, TMTextUnitCurrentVariant> tmTextUnitCurrentVariantCache = importerCacheService.createTmTextUnitCurrentVariantCache();
+
+        logger.debug("Convert to ThirdPartyTextUnitForBatchImport");
+        return thirdPartyTextUnitDTOSet.stream()
+                .filter(logIfFalse(t -> t.getThirdPartyTextUnitId() != null, logger, "Missing mandatory third party text unit id, skip: {}", ThirdPartyTextUnitDTO::getThirdPartyTextUnitId))
+                .filter(logIfFalse(t -> t.getMappingKey() != null, logger, "Missing mandatory mapping key, skip: {}", ThirdPartyTextUnitDTO::getThirdPartyTextUnitId))
+
+                .map(t -> {
+                    ThirdPartyTextUnitForBatchImport thirdPartyTextUnitForBatchImport = new ThirdPartyTextUnitForBatchImport();
+                    thirdPartyTextUnitForBatchImport.setThirdPartyTextUnitId(t.getThirdPartyTextUnitId());
+                    thirdPartyTextUnitForBatchImport.setMappingKey(t.getMappingKey());
+
+                    thirdPartyTextUnitForBatchImport.setRepository(repositoriesCache.getUnchecked(t.getRepositoryName()));
+                    if (thirdPartyTextUnitForBatchImport.getRepository() != null) {
+                        thirdPartyTextUnitForBatchImport.setAsset(assetsCache.getUnchecked(new AbstractMap.SimpleEntry<>(
+                                t.getAssetPath(), thirdPartyTextUnitForBatchImport.getRepository().getId())));
+
+                        if (thirdPartyTextUnitForBatchImport.getAsset() != null) {
+                            TMTextUnitCurrentVariant foundCurrentVariant;
+                            if (isPluralFile) {
+                                foundCurrentVariant = tmTextUnitCurrentVariantCache.getUnchecked(
+                                        new AbstractMap.SimpleEntry<>(
+                                                t.getTmTextUnitName() + pluralSuffix,
+                                                thirdPartyTextUnitForBatchImport.getAsset().getId())
+                                );
+                            } else {
+                                foundCurrentVariant = tmTextUnitCurrentVariantCache.getUnchecked(
```

Comment:
Trying to guess since the code calling it is not available but is this going to do 1 query per smartling string?

Reply:
Yes, its a map called on the set stream so the query is 1 per string